src/main/java/dk/cngroup/lentils/service/ImageService.java

```diff
@@ -89,13 +90,33 @@ public Resource loadAsResource(final String filename) {
         }
     }
 
-    public String getFilePathToSave(final MultipartFile file) {
+    private String getFileName(final MultipartFile file) {
         if (file.isEmpty()) {
             throw new FileNotFoundException();
         }
-        String fileName = file.getOriginalFilename();
-        FileUtils.createDirectory(getDirectory());
-        return getDirectory() + fileName;
+        return addHashToFilename(file.getOriginalFilename());
+    }
+
+    public String addHashToFilename(final String filename) {
+        int i = filename.lastIndexOf('.');
+        if (i < 0) {
```

Comment:
Celkove by pak bylo hezci, kdyby jsi zavolala tu metodu `generateHash` jen jednou - viz nasledujici koment.
Takze by to bylo nejak takto:
```java
String uniqueSuffix = generateHash();
if (i < 0) {
    ... // use variable uniqueSuffix
} else {
    ... // use variable uniqueSuffix
}

Mozna bych taky metodu `addHashToFilename` prejmenoval na `addUniquePostfixToFilename` a metodu `generateHash` na `generateUniqueString` protoze krome hashe se tam pridava i datum.